src/main/java/com/cloudbees/jenkins/support/startup/StartupComponent.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.support.startup;
+
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.PrefilteredPrintedContent;
+import com.cloudbees.jenkins.support.filter.ContentFilter;
+import com.cloudbees.jenkins.support.timer.UnfilteredFileListCapComponent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.Util;
+import hudson.init.InitMilestone;
+import java.io.PrintWriter;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Component that contributes the startup report into the support bundle.
+ */
+@Extension
+public class StartupComponent extends UnfilteredFileListCapComponent {
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Startup Report";
+    }
+
+    @Override
+    public void addContents(@NonNull Container container) {
+        super.addContents(container, StartupReport.get().getLogs());
+        container.add(new StartupContent(StartupReport.get().getTimesPerMilestone()));
+    }
+
+    @NonNull
+    @Override
+    public ComponentCategory getCategory() {
+        return ComponentCategory.CONTROLLER;
+    }
+
+    private static class StartupContent extends PrefilteredPrintedContent {
+        public static final String MILESTONE_HEADER = "Milestone";
+        public static final String TIME_HEADER = "Time";
+        private final Map<InitMilestone, Long> timePerMilestone;
+
+        StartupContent(Map<InitMilestone, Long> timePerMilestone) {
+            super("startup-timings.md");
+            this.timePerMilestone = timePerMilestone;
+        }
+
+        @Override
+        protected void printTo(PrintWriter out, @NonNull ContentFilter filter) {
+            var startTime = ManagementFactory.getRuntimeMXBean().getStartTime();
```

Comment:
Beware that this is _JVM_ startup time, so if any servlet listener runs before Jenkins starts (two such examples come to mind in CloudBees CI), you will get a very skewed measurement.